app/serializers/v1/fee_serializer.rb

```diff
@@ -52,6 +52,18 @@ def serialize
     private
 
     def date_boundaries
+      if model.charge? && !model.pay_in_advance? && model.charge.pay_in_advance?
+        subscription = model.subscription
+        invoice = model.invoice
+        timestamp = invoice.invoice_subscription(subscription.id).timestamp
+        interval = invoice.charge_pay_in_advance_interval(timestamp, subscription)
+
+        return {
+          from_date: interval[:charges_from_date]&.to_datetime&.iso8601,
```

Comment:
On a global perspective, I'm wondering if we should store the correct boundaries in the fee (or in new attributes `charge_in_advance_from|to_datetime` of the invoice subscription)

Reply:
Yep that is a good question, but since pay_in_advance charges can be recurring, existing boundaries are correct in terms of aggregation (since for recurring that are pay in advance we still need to generate invoice at the end of the period and use existing boundaries). But in terms of UI / api payload presentation, those are not correct. Do you think we should have both persisted?

Reply:
Yes, that was my point. Having to recompute it each time, makes me think we should store it, mainly for performance and maintainability (pretty sure we will forget to use the right boundaries in some other places)

Reply:
Writing this task in my to-do list